packages/client/MessagePickupRepositoryClient.ts

```diff
@@ -0,0 +1,316 @@
+import { Client } from 'rpc-websockets'
+import { Logger } from '@nestjs/common'
+import {
+  JsonRpcParamsMessage,
+  RemoveAllMessagesOptions,
+  ConnectionIdOptions,
+  AddLiveSessionOptions,
+} from './interfaces'
+import {
+  AddMessageOptions,
+  GetAvailableMessageCountOptions,
+  MessagePickupRepository,
+  QueuedMessage,
+  RemoveMessagesOptions,
+  TakeFromQueueOptions,
+} from '@credo-ts/core'
+
+export class MessagePickupRepositoryClient implements MessagePickupRepository {
+  private client: Client
+  private readonly logger = new Logger(MessagePickupRepositoryClient.name)
+  private messageReceivedCallback: ((data: JsonRpcParamsMessage) => void) | null = null
+
+  constructor(private readonly url: string) {}
```

Comment:
Maybe we can set the messageReceivedCallback directly on the constructor, so we don't need a separate method? Or it is needed to define it after constructing it?

Reply:
Although I don’t see any issue with defining the callback when instantiating the class to perform this task, I’m thinking of renaming the connect method to initialize and calling it from the constructor. Additionally, I would pass the Agent as a parameter so it can handle the logic to dispatch messages. What do you think? Does it sound good to you?